### DISCUSSION TITLE
VS Error not highlighting

### DISCUSSION THREAD

**Discussion by Edacheclarus2** [Cyfrin/foundry-full-course-cu#583] [General]
I'm currently facing an issue on VS Code: errors not highlighting in real-time,  only showing after run/compile. Is this normal? 

Also, I need a  recommendation on  Copilot  alternative or  auto-complete extension that's free to use, because solidity visual developer extension is not highlighting or helping in auto completing. 

Someone help me.
Thanks 🙏

**Comment by shikhar229169** [comment]
Yea, I also faced that issue when the errors were not highlighted, I had to restart my vs code 2-3 times and the issue was fixed.

I use github copilot, actually I have the guthub student developer pack so its free to use. 
So, you can get student developer by registering here: https://education.github.com/pack
And once it gets verified you can avail copilot services: https://github.com/settings/copilot

**Reply by Edacheclarus2** [reply]
I'm still facing challenge applying for the student developer pack. I really need Co pilot

Thank you for the response

**Comment by elec7rik** [comment]
hi @Edacheclarus2 for the error highlighting problem- ig you should reinstall the vscode extensions that patrick shows in the lesson.
for the github copilot issue- i was in a similar condition a few weeks ago, it took a week for me to get the benefits after the application being approved. but i disabled the access to copilot in github anyway after getting it, since it types in the whole code at once and it makes retaining things too hard. didn't work for me personally but you should do as you like.

**Reply by Edacheclarus2** [reply]
Yes, this worked.

I have to close and reopen my VSCode.

About, the Copilot, I'm still not using it. Student park not being accessible still, and for now, I'm trying to cope with getting used to ChatGpt @elec7rik 

Thanks for your input and replies. 🙏
